spec/controllers/spree/credit_cards_controller_spec.rb

```diff
@@ -69,6 +69,53 @@
     end
   end
 
+  describe "#update" do
+    let(:params) { { format: :json, credit_card: { is_default: true } } }
+    context "when the specified credit card is not found" do
+      before { params[:id] = 123 }
+
+      it "renders a flash error" do
+        put :update, params
+        json_response = JSON.parse(response.body)
+        expect(json_response['flash']['error']).to eq I18n.t(:card_could_not_be_updated)
+      end
+    end
+
+    context "when the specified credit card is found" do
+      let!(:card) { create(:credit_card, gateway_customer_profile_id: 'cus_AZNMJ') }
+      before { params[:id] = card.id }
+
+      context "but the card is not owned by the user" do
+        it "redirects to unauthorized" do
+          put :update, params
+          expect(response).to redirect_to spree.unauthorized_path
+        end
+      end
+
+      context "and the card is owned by the user" do
+        before { card.update_attribute(:user_id, user.id) }
+
+        context "when the update completes successfully" do
+          it "renders a serialized copy of the updated card" do
+            expect{ put :update, params }.to change { card.reload.is_default }.to(true)
```

Comment:
Just a sanity check while I was working on the code. I can delete it you think it is messy, but I don't think it is likely to cause us any problems...

Reply:
To me it's more about the maintenance cost this lines will have just because we weren't 100% confident when writing the test. I'd like to hear other opinions though.